packages/cardano-services/test/Asset/TypeormAssetProvider.test.ts

```diff
@@ -0,0 +1,171 @@
+import { Cardano, ProviderError } from '@cardano-sdk/core';
+import {
+  TypeOrmNftMetadataService,
+  TypeormAssetProvider,
+  createDnsResolver,
+  getConnectionConfig,
+  getEntities
+} from '../../src';
+import { logger } from '@cardano-sdk/util-dev';
+import { sleep } from '../util';
+
+const testHandleAssetId = Cardano.AssetId(
+  '62173b90b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a48656c6c6f48616e646c65'
+);
+const noMetadataAssetId = Cardano.AssetId(
+  '62173b90b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a283232322968616e646c653638'
+);
+const randomHandleAssetId = Cardano.AssetId(
+  '62173b90b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a446f75626c6548616e646c65'
+);
+
+const testHandleAssetInfo = {
+  assetId: testHandleAssetId,
+  fingerprint: 'asset13xe953tueyajgxrksqww9kj42erzvqygyr3phl',
+  name: '48656c6c6f48616e646c65',
+  policyId: '62173b90b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a',
+  quantity: 1n,
+  supply: 1n
+};
+
+const noMetadataAssetInfo = {
+  assetId: noMetadataAssetId,
+  fingerprint: 'asset1ju4qkyl4p9xszrgfxfmu909q90luzqu0nyh4u8',
+  name: '283232322968616e646c653638',
+  policyId: '62173b90b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a',
+  quantity: 1n,
+  supply: 1n
+};
+
+const randomHandleAssetInfo = {
+  assetId: randomHandleAssetId,
+  fingerprint: 'asset1ss4nvcah07l2492qrfydamvukk4xdqme8k22vv',
+  name: '446f75626c6548616e646c65',
+  policyId: '62173b90b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a',
+  quantity: 2n,
+  supply: 2n
+};
+
+const testHandleOtherProperties = new Map<string, string | Array<string> | Map<string, string | bigint>>([
+  ['augmentations', []],
+  [
+    'core',
+    new Map<string, string | bigint>([
+      ['handleEncoding', 'utf-8'],
+      ['og', 0n],
+      ['prefix', '$'],
+      ['termsofuse', 'https://cardanofoundation.org/en/terms-and-conditions/'],
+      ['version', 0n]
+    ])
+  ],
+  ['website', 'https://cardano.org/']
+]);
+
+const testHandleNftMetadata = {
+  description: 'The Handle Standard',
+  image: 'ipfs://some-hash',
+  name: 'HelloHandle',
+  otherProperties: testHandleOtherProperties
+};
+
+const randomHandleNftMetadata = {
+  description: 'The Handle Standard',
+  image: 'ipfs://some-hash',
+  name: 'DoubleHandle',
+  otherProperties: testHandleOtherProperties
+};
+
+describe('TypeormAssetProvider', () => {
```

Comment:
Please cross-reference `DbSyncAssetProvider.test.ts` to make sure we're not missing some test case(s)

Reply:
Yes. Test cases are covered.

Reply:
I think we're missing tests for resolving `tokenMetadata`

Reply:
But I didn't add getting `tokenMetadata` to the TypeormAssetProvider. 🤔

Reply:
Please add it, otherwise we can't use this provider as it's incomplete

Reply:
Added on 1509c1e